src/NuGet.Core/NuGet.Packaging/NupkgMetadata/NupkgMetadataFileFormat.cs

```diff
@@ -57,6 +57,11 @@ public static NupkgMetadataFile Read(TextReader reader, ILogger log, string path
                     Strings.Error_LoadingHashFile,
                     path, ex.Message));
 
+                if (!string.IsNullOrEmpty(path) && ex is InvalidDataException)
+                {
+                    throw new InvalidDataException(message: path, innerException: ex);
```

Comment:
I don't think that having the exception message be just the filename without anything else is the best experience. Consider ArgumentNullException in the BCL. The exception message isn't just the argument name. Similarly, `FileNotFoundException`'s exception message is "Could not find file 'c:\does not exist.txt'.", not just the filename.

```suggestion
                    throw new InvalidDataException(message: $"Error reading 'path'.", innerException: ex);
```

However, since that message gets surfaced to customers, it probably needs to go into strings.resx.

Reply:
I see. I felt that a path and "invalid data" as an exception was pretty straight forward. I can introduce some localized strings.
Your example states there was an error reading the file. Is that intuitive to mean that the file's data was corrupt/not expected, or the contents were empty?